File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var increasingTriplet = function(nums) {
  let first = Infinity;
  let second = Infinity;
  
  for (let i = 0; i < nums.length; i++) {
    const cur = nums[i];
    
    if (cur > first && cur > second) {
      return true;
    }
    
    if (cur > first) {
      second = cur;
    } else {
      first = cur;
    }
  }
  
  return false;
};

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        first = float("inf")
        second = float("inf")

        for num in nums:
            if num > first and num > second: return True
            if num > first: second = num
            else: first = num

        return False
